### PR TITLE
Memgraph platform

### PR DESCRIPTION
**Pull request by tewnut**
- Add memgraph-platform chart.
- Upgrade memgraph chart default version to 2.13.0

### CODE REVIEW

**Comment by antejavor** [comment]
Hi @tewnut, thanks a lot for opening a PR 🚀 . I will look into this in the next few days!

File: charts/memgraph-platform/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+| `readinessProbe`                             | Readiness probe settings                                    | `{}`                        |
+| `autoscaling.enabled`                        | Enable Horizontal Pod Autoscaler                            | `false`                     |
+| `autoscaling.minReplicas`                    | Minimum number of replicas for autoscaling                  | `1`                         |
+| `autoscaling.maxReplicas`                    | Maximum number of replicas for autoscaling                  | `100`                       |
+| `autoscaling.targetCPUUtilizationPercentage` | Target CPU utilization percentage for autoscaling           | `80`                        |
+| `volumes`                                    | Additional volumes on the output Deployment definition      | `[]`                        |
+| `volumeMounts`                               | Additional volumeMounts on the output Deployment definition | `[]`                        |
+| `nodeSelector`                               | Node selector for pods                                      | `{}`                        |
```
Comment (CarlEkerot) [inline charts/memgraph-platform/README.md:71]: `nodeSelector`, `tolerations` and `affinity` doesn't seem to be used in the chart.

**Comment by CarlEkerot** [comment]
Been using this somewhat successfully the last week or so. I am not knowledgeable enough in Helm charts to give a proper review, but here are some pointers as a user:

* Deploying this from IaC, it would be nice to (optionally) provide existing volume claims for storage
* I didn't manage to get the ingress to work properly. Haven't spent too much time debugging the issue.
* See review comment regarding `nodeSelector`, `tolerations` and `affinity`

**Comment by karmenrabar** [comment]
Hello @CarlEkerot, thanks so much for the review! @antejavor is currently on his AL, but he'll get back to you as soon as he's back. I'm glad to hear that it's going successfully for you. Could you tell us what you are working on? :)

**Comment by CarlEkerot** [comment]
> Hello @CarlEkerot, thanks so much for the review! @antejavor is currently on his AL, but he'll get back to you as soon as he's back. I'm glad to hear that it's going successfully for you. Could you tell us what you are working on? :)

Mostly poking around with memgraph and a few other graph DBs :) Looking forward to using this chart once it's in a stable state :+1:

**Comment by karmenrabar** [comment]
Nice to hear that you're exploring Memgraph, @CarlEkerot  :) Feel free to ask if you have more questions, and don't hesitate to reach out on our [discord](https://discord.com/invite/memgraph) or book an [office hours call ](https://calendly.com/d/3jv-347-54b/memgraph-office-hours)!

**Comment by tewnut** [comment]
> Been using this somewhat successfully the last week or so. I am not knowledgeable enough in Helm charts to give a proper review, but here are some pointers as a user:
> 
> * Deploying this from IaC, it would be nice to (optionally) provide existing volume claims for storage
> * I didn't manage to get the ingress to work properly. Haven't spent too much time debugging the issue.
> * See review comment regarding `nodeSelector`, `tolerations` and `affinity`

can you share the ingress section of your values.yaml file?

**Comment by katarinasupe** [comment]
Hi @tewnut, thank you so much for providing the chart for the Memgraph Platform. With the Memgraph Lab image being available on the Docker hub now, we have been working on a new way of running the Memgraph Platform that should ease the deployment process, and that's why we still need to review this. With Memgraph 2.15, we will make these changes public, and then we'll see what to do with Helm Charts (how to approach the Memgraph Platform deployment properly). 

@CarlEkerot thank you for the review as well! 👏  Stay tuned for the updates 😄

**Comment by tewnut** [comment]
> Hi @tewnut, thank you so much for providing the chart for the Memgraph Platform. With the Memgraph Lab image being available on the Docker hub now, we have been working on a new way of running the Memgraph Platform that should ease the deployment process, and that's why we still need to review this. With Memgraph 2.15, we will make these changes public, and then we'll see what to do with Helm Charts (how to approach the Memgraph Platform deployment properly).
> 
> @CarlEkerot thank you for the review as well! 👏 Stay tuned for the updates 😄

Thanks, looking forward to the new release...

**Review by antejavor — CHANGES_REQUESTED** [review]
Hi @tewnut,


First of all, thanks for the contribution and effort. 🚀 

As @katarinasupe [has said previously](https://github.com/memgraph/helm-charts/pull/13#issuecomment-1961529626) and 
as you can read in [release notes](https://memgraph.com/docs/release-notes#transition-to-multi-container-memgraph-platform-from-version-215), we have changed how Memgraph platform works. 

Now, it is a Docker compose (based on the [Memgraph mage](https://hub.docker.com/r/memgraph/memgraph-mage/tags) and [Memgraph lab](https://hub.docker.com/r/memgraph/lab). Those are two separate containers now; there is no longer a single container image that has both products. 

Unfortunately, this means this chart is not valid. Approximately three months ago, when you opened PR, we started planning the transition to the new platform. Due to upcoming changes and other priorities, this PR didn't receive proper attention. 

As you can see in our [roadmap](https://github.com/memgraph/roadmap), we started working on improving our Kubernetes experience, and this repository will receive more changes and updates in the following weeks. 

One of the updates is to create the helm charts for replication and high-availability support.  

This will be my main focus, and having a platform chart would be a great addition to this repository. 

The question is, are you willing to adopt this chart to base the deployment on the two images (mage and lab) so we can wrap up and merge this PR?

**Comment by tewnut** [comment]
Yeah, I think this is altogether better approach. Thanks for the update. 

Let me try creating the charts for my k8s cluster. When things are in good shape, I will create a PR in this repo, should be a couple of days...

**Comment by antejavor** [comment]
> Yeah, I think this is a better approach altogether. Thanks for the update.
> 
> Let me try creating the charts for my k8s cluster. When things are in good shape, I will create a PR in this repo, should be a couple of days...

Keep in mind we updated the contribution guide a bit: https://github.com/memgraph/helm-charts/commit/c1cc60603a8ea2f3572e5f5685190b65b60b7c7b 

If you are going to create a new PR, please use the latest release branch and create a PR on a specific release branch. 

If you are going to use this PR it is also fine.

**Comment by antejavor** [comment]
Hi @tewnut can we close this PR now?

**Pull request closed without merging**